### ISSUE TITLE
Append or prepend selected text to your clipboard with different separators

### ISSUE THREAD

**Issue by bigplayer-ai** [p0deje/Maccy#816]
### Before Submitting Your Feature Request

- [X] Check that there isn't already a similar feature request to avoid creating a duplicate.
- [X] I have seen the [FAQ](https://github.com/p0deje/Maccy?tab=readme-ov-file#faq).

### Problem

Append or prepend selected text to your clipboard with different separators

### Solution

Append Clipboard
Append or prepend selected text to your clipboard with different separators

**Comment by p0deje** [comment]
i think that's similar to #239, please provide more details if it is not.

**Comment by bigplayer-ai** [comment]
> i think that's similar to #239, please provide more details if it is not.

That’s not correct. You can use keyboard shortcuts like Command + Alt + C to append items to the clipboard with a space separator (can be changed via settings). For instance, if you copy “Google” using Command + C, and then copy “Safari” using Command + Alt + C, your clipboard will contain “Google Safari”.

**Comment by p0deje** [comment]
> That’s not correct. You can use keyboard shortcuts like Command + Alt + C to append items to the clipboard with a space separator (can be changed via settings). For instance, if you copy “Google” using Command + C, and then copy “Safari” using Command + Alt + C, your clipboard will contain “Google Safari”.

So you want Maccy to implement this feature? What's the use case, why would you want such a feature?

**Comment by bigplayer-ai** [comment]
> > That’s not correct. You can use keyboard shortcuts like Command + Alt + C to append items to the clipboard with a space separator (can be changed via settings). For instance, if you copy “Google” using Command + C, and then copy “Safari” using Command + Alt + C, your clipboard will contain “Google Safari”.
> 
> So you want Maccy to implement this feature? What's the use case, why would you want such a feature?

Implementing this feature in Maccy would greatly improve workflows by allowing users to quickly and efficiently combine multiple pieces of information.

The current solution using macOS shortcuts is **cumbersome and inefficient**, especially when copying more than 10 items, such as an address. This new feature would eliminate the need for manual concatenation, saving users time and boosting productivity.

User requests, like issue #239, highlight the limitations of existing methods and the high demand for this functionality. By enabling seamless aggregation/concatenations of clipboard items, Maccy would provide a superior user experience, streamlining tasks like research, writing, and coding.

This feature is highly requested and would significantly improve workflows and productivity by addressing the shortcomings of the current method. Issue #239 reflects a similar need for better clipboard management.

So it is more like batch copying in one clipboard item instead of sequential pasting, this way it would be more organized clipboard manager with additional flexibility.

**Comment by p0deje** [comment]
Have you generated most of the text using AI? 😄 

I understand what you request, I'm interested in learning the _exact use case_ you would use this feature for. Please, elaborate on that, because I would not want to implement a feature that is based on an abstract need.

**Comment by bigplayer-ai** [comment]
> Have you generated most of the text using AI? 😄
> 
> I understand what you request, I'm interested in learning the _exact use case_ you would use this feature for. Please, elaborate on that, because I would not want to implement a feature that is based on an abstract need.

**I need to copy scattered items on a page with a specific delimiter.**

For example, when copying a client's address, I want to select the street address, name, city, and zip code in my desired order, all separated by a comma. 

---

This feature would also be useful for coding tasks, such as:
- Copying multiple strings from the internet into a CSV file
- Gathering scattered words from Wikipedia or Google

Additionally, it would help when formulating specific terms for my research from multiple pieces on a page.

**Comment by p0deje** [comment]
Rather than trying to append/prepend a new copy to an existing one and having to remember different shortcuts, what if Maccy allows you to select multiple items in history (e.g. hold Shift and press the down arrow to select Address, City, State, ZIP). Then, when you press Enter it asks how these copies should be joined together (e.g. separated by a newline or comma). Then, depending on your settings, Maccy would either create a new copy or paste it immediately.

Would this satisfy your use cases? I plan to implement something similar as part of #239

**Comment by p0deje** [comment]
You don't need to have Maccy or anything except the standard Shortcuts app to do this. Here is an example of a workflow that you can use to create a new copy that consists of the current copy + new copy joined with a newline - https://www.icloud.com/shortcuts/6474f3ed1d0e47aa9caf2bd4087fec1e. The default shortcut is Option+Command+C.

<img width="1312" alt="Screenshot 2024-11-27 at 21 42 12" src="https://github.com/user-attachments/assets/c80eb723-297f-4a21-874d-0cb951116ee1">

**Comment by bigplayer-ai** [comment]
> You don't need to have Maccy or anything except the standard Shortcuts app to do this. Here is an example of a workflow that you can use to create a new copy that consists of the current copy + new copy joined with a newline - https://www.icloud.com/shortcuts/6474f3ed1d0e47aa9caf2bd4087fec1e. The default shortcut is Option+Command+C.
> 
> <img alt="Screenshot 2024-11-27 at 21 42 12" width="1312" src="https://private-user-images.githubusercontent.com/665846/390654206-c80eb723-297f-4a21-874d-0cb951116ee1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3NzkyNjUsIm5iZiI6MTczMjc3ODk2NSwicGF0aCI6Ii82NjU4NDYvMzkwNjU0MjA2LWM4MGViNzIzLTI5N2YtNGEyMS04NzRkLTBjYjk1MTExNmVlMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyOFQwNzI5MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NGE2MzBhNDM0YjI2ZjliOTcxMGE2Zjc1YTJiZTkzY2JhYmM1NDlkOTQ0OGFmMzkwOTQzNDBkMDVkZDZjZDZhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.EEebeBXridjuPWpaOTXbCDdh7SsoM-13nBiHn4CPQkg">

Thanks for the shortcut🙏 

I actually use a Raycast extension that Achieves the same functionality https://www.raycast.com/koinzhang/append-clipboard

**Comment by p0deje** [comment]
> There is a raycast extension that does that: "https://www.raycast.com/koinzhang/append-clipboard", but I don't use raycast.

I guess this changed now 🙂
I'm going to close this for now since Shorcut is a decent solution.
